tests/genericmetadata_test.py

```diff
@@ -22,6 +23,78 @@ def test_metadata_overlay(md: comicapi.genericmetadata.GenericMetadata, replaced
     assert md == expected
 
 
+def test_metadata_overlay_overwrite():
+    md = comicapi.genericmetadata.GenericMetadata(series="test", issue="1", title="test", genres={"test", "test2"})
+    overwrite = comicapi.genericmetadata.GenericMetadata(
+        series="test2", issue="2", title="test2", genres={"test3", "test4"}
+    )
+    md.overlay(overwrite, OverlayMode.overwrite)
+
+    assert md == overwrite
+
+
+def test_metadata_overlay_overwrite_no_clear():
+    md = comicapi.genericmetadata.GenericMetadata(
+        series="test", issue="1", title="test", genres={"test", "test2"}, story_arcs=["arc"]
+    )
+    overwrite = comicapi.genericmetadata.GenericMetadata(
+        series="test2", issue="2", title="test2", genres=set(), story_arcs=[]
+    )
+    expected = comicapi.genericmetadata.GenericMetadata(
+        series="test2", issue="2", title="test2", genres={"test", "test2"}, story_arcs=["arc"]
+    )
+    md.overlay(overwrite, OverlayMode.overwrite_no_clear)
+
+    assert md == expected
```

Comment:
This test needs to show how it is different from overlay. I don't think it ever makes sense to use `overwrite_no_clear` as it is written

Reply:
Now it should show how it's different with comes back to the commandline clear again

Reply:
This doesn't show all of the types in play.
```python
def test_metadata_overlay_overwrite_no_clear():
    md = comicapi.genericmetadata.GenericMetadata(
        series="test", title="test", issue="1", volume_count=1, page_count=3, day=3, genres={"test", "test2"}, story_arcs=["arc"], characters={"bob", "fred"}
    )
    overwrite = comicapi.genericmetadata.GenericMetadata(
        series="test2", title="", issue_count=2, page_count=2, day=0, genres={'fake'}, characters="", scan_info="nothing",
    )
    expected = comicapi.genericmetadata.GenericMetadata(
        series="test2",
        title="test",
        issue="1",
        issue_count=2,
        volume_count=1,
        day=0,
        page_count=2,
        genres={"fake"},
        story_arcs=["arc"],
        characters={"bob", "fred"},
        scan_info="nothing",
    )
    md.overlay(overwrite, OverlayMode.overwrite_no_clear)

    assert md == expected
```

With this in place we test far more.

- If a `str` is set on both the second value is used: `series`
- If a `str` is set on the first and on the second an empty string is set the first is used: `title`
- If a `str` is set on the first and on the second it is __not__ set the first is used: `issue`
- If a `str` is __not__ set on the first and on the second it is set the second is used: `scan_info`
- If a `int` is set on the first and on the second it is __not__ set the first is used: `volume_count`
- If a `int` is set on both the second value is used: `page_count`
- If a `int` is __not__ set on the first and on the second it is set the second is used: `issue_count`
- If a `int` is set on the first and on the second a `0` is set the second is used: `day`
- If a `set` is set on both the second value is used: `genres`
- If a `list` is set on the first and on the second it is __not__ set the first is used: `story_arcs`
- If a `set` is set on the first and on the second an empty string is set the first is used: `characters`

Now that there is a more complete test, where would this be used? Unless I am mistaken all of these cases are taken care of elsewhere in comictagger. In taggerwindow.py `form_to_metadata` utils.xlate is used to ensure empty string values are set to `None` the same is done with numbers except for `critical_rating` which is special cased at the value 0.0 to be set to `None` because the control doesn't allow it being unset. The same is used in comicrack.py `_convert_xml_to_metadata` except for the special case for `critical_rating` (which means that if someone explicitly set the value to 0 ComicTagger cannot handle it properly and will unset it)